skfp/datasets/tdc/benchmark.py

```diff
@@ -0,0 +1,332 @@
+        is used, by default `$HOME/scikit_learn_data`.
+
+    as_dict : bool, default=False
+        If True, returns the splits as dictionary with keys "train", "valid" and "test",
+        and index lists as values. Otherwise, returns three lists with splits indexes.
+
+    verbose : bool, default=False
+        If True, progress bar will be shown for downloading or loading files.
+
+    Returns
+    -------
+    data : tuple(list[int], list[int], list[int]) or dict
+        Depending on the `as_dict` argument, one of:
+        - three lists of integer indexes
+        - dictionary with "train", "valid" and "test" keys, and values as lists with
+        splits indexes
+
+    """
+    splits = fetch_splits(
+        data_dir,
+        dataset_name=f"TDC_{dataset_name}",
+        filename=f"tdc_{dataset_name.lower()}_splits.json",
+        verbose=verbose,
+    )
+    if as_dict:
+        return splits
+    else:
+        return splits["train"], splits["valid"], splits["test"]
+
+
+def _subset_to_dataset_names(subset: Union[str, list[str], None]) -> list[str]:
+    # transform given subset (e.g. "adme", "hts" or "tox") into list of dataset names
+    # for appropriate TDC datasets
+
+    adme_names = [
+        "approved_pampa_ncats",
+        "b3db_classification",
+        "b3db_regression",
+        "bioavailability_ma",
+        "caco2_wang",
+        "clearance_hepatocyte_az",
+        "clearance_microsome_az",
+        "cyp1a2_veith",
+        "cyp2c19_veith",
+        "cyp2c9_substrate_carbonmangels",
+        "cyp2c9_veith",
+        "cyp2d6_substrate_carbonmangels",
+        "cyp2d6_veith",
+        "cyp3a4_substrate_carbonmangels",
+        "cyp3a4_veith",
+        "half_life_obach",
+        "hia_hou",
+        "hlm",
+        "pampa_ncats",
+        "pgp_broccatelli",
+        "ppbr_az",
+        "rlm",
+        "solubility_aqsoldb",
+        "vdss_lombardo",
+    ]
+
+    hts_names = [
+        "sarscov2_3clpro_diamond",
+        "sarscov2_vitro_touret",
+    ]
+
+    tox_names = [
+        "ames",
+        "carcinogens_lagunin",
+        "dili",
+        "herg",
+        "herg_central_at_10um",
+        "herg_central_at_1um",
+        "herg_central_inhib",
+        "herg_karim",
+        "ld50_zhu",
+        "skin_reaction",
+    ]
+
+    all_dataset_names = adme_names + hts_names + tox_names
+
+    if subset is None:
+        dataset_names = all_dataset_names
+    elif subset == "adme":
+        dataset_names = adme_names
+    elif subset == "hts":
+        dataset_names = hts_names
+    elif subset == "tox":
+        dataset_names = tox_names
+    elif isinstance(subset, (list, set, tuple)):
+        for name in subset:
+            if name not in all_dataset_names:
+                raise ValueError(
+                    f"Dataset name '{name}' not recognized among TDC datasets"
+                    f"Some TDC datasets can be imported from MoleculeNet benchmark"
```

Comment:
Just `f"Dataset name '{name}' not recognized among TDC datasets"`, for uniformity with MoleculeNet